### DISCUSSION TITLE
How to label an unnamed dimension to access it

### DISCUSSION THREAD

**Discussion by wmcfarlane-smith** [SciTools/iris#5422] [Q&A] [answered]
Some data I have doesn't have the time coordinate labelled. It looks like this:
`CLOUD CONDENSED ICE COLUMN / (kg/m^2) (-- : 600; latitude: 90; longitude: 144)`
`    Dimension coordinates:`
`        latitude                          -              x              -`
`        longitude                         -              -              x`

The coordinate with 600 points is time but as it has no name I can't access it with `cube_1.coords()`, only longitude and latitude appear. Is there a way to name time so I can access it? Thanks

**Comment by trexfeathers** [comment, marked as answer]
Hi @wmcfarlane-smith

The labelling you refer to is the presence of a **dimension coordinate** on that dimension (more info here: [Iris Data Structures](https://scitools-iris.readthedocs.io/en/stable/userguide/iris_cubes.html)). The `--` means this is an 'anonymous dimension' - there is no coordinate at all for this dimension. An un-named coordinate for this dimension would instead show as `unknown`.

You are free to add your own coordinates to an existing `Cube`, with the following two steps:

1. Create your dimension coordinate. At minimum you will need to include some points for the coordinate, and a name - the name that will appear on the `Cube` summary.
   For a time coordinate I would also recommend assigning `units`, to give the numbers some meaning. This can either be a string, like the example below, or a cf-units [`Unit`](https://cf-units.readthedocs.io/en/stable/unit.html).
3. Add your new dimension coordinate to your `Cube`.

```python
from iris.coords import DimCoord
import numpy as np


# Learn more about valid units: https://cf-units.readthedocs.io/en/stable/unit.html
time_coord = DimCoord(np.arange(600), units="days since 1970-01-01")
# Using rename will intelligently assign to the standard_name or long_name property, as appropriate.
time_coord.rename("time")

# 0 refers to which dimension you are adding the coordinate to.
cube_1.add_dim_coord(time_coord, 0)

```

**Comment by wmcfarlane-smith** [comment]
Hi Martin,

That's really helpful. Thank you!
________________________________
